sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -752,29 +732,21 @@ class Analyzer(
     //    and the default database is only used to look up a view);
     // 3. Use the currentDb of the SessionCatalog.
     private def lookupTableFromCatalog(
-        tableIdentifier: TableIdentifier,
+        ident: Seq[String],
         u: UnresolvedRelation,
         defaultDatabase: Option[String] = None): LogicalPlan = {
-      val tableIdentWithDb = tableIdentifier.copy(
-        database = tableIdentifier.database.orElse(defaultDatabase))
+      val identWithDb = if (ident.length == 1) {
+        defaultDatabase.toSeq ++ ident
+      } else {
+        ident
+      }
       try {
-        catalog.lookupRelation(tableIdentWithDb)
+        catalog.lookupRelation(identWithDb)
       } catch {
         case _: NoSuchTableException | _: NoSuchDatabaseException =>
           u
       }
     }
-
-    // If the database part is specified, and we support running SQL directly on files, and
```

Comment:
This is not needed anymore because in https://github.com/apache/spark/pull/24741 we delay the error reporting of unresolved relation to `CheckAnalysis`